src/mlpack/methods/ann/layer/max_pooling.hpp

```diff
@@ -56,19 +58,58 @@ class MaxPooling
   MaxPooling();
 
   /**
-   * Create the MaxPooling object using the specified number of units.
+   * Create the MaxPooling object using the specified number of units and padding.
    *
    * @param kernelWidth Width of the pooling window.
    * @param kernelHeight Height of the pooling window.
    * @param strideWidth Width of the stride operation.
    * @param strideHeight Width of the stride operation.
    * @param floor Rounding operator (floor or ceil).
+   * @param inputWidth Width of input.
+   * @param inputHeight Height of input.
+   * @param padW Padding width of the input.
+   * @param padH Padding height of the input.
+   * @param paddingType The type of padding (Valid or Same). Defaults to None.
    */
   MaxPooling(const size_t kernelWidth,
              const size_t kernelHeight,
              const size_t strideWidth = 1,
              const size_t strideHeight = 1,
-             const bool floor = true);
+             const bool floor = true,
+             const size_t inputWidth = 0,
+             const size_t inputHeight = 0,
+             const size_t padW = 0,
+             const size_t padH = 0,
+             const std::string paddingType = "None");
+
+  /**
+   * Create the MaxPooling object using the specified number of units and padding.
+   *
+   * @param kernelWidth Width of the pooling window.
+   * @param kernelHeight Height of the pooling window.
+   * @param strideWidth Width of the stride operation.
+   * @param strideHeight Width of the stride operation.
+   * @param floor Rounding operator (floor or ceil).
+   * @param inputWidth Width of input.
+   * @param inputHeight Height of input.
+   * @param padW A two-value tuple indicating padding widths of the input.
+   *             First value is padding at left side. Second value is padding on
+   *             right side.
+   * @param padH A two-value tuple indicating padding heights of the input.
+   *             First value is padding at top. Second value is padding on
+   *             bottom.
+   * @param paddingType The type of padding (Valid or Same). Defaults to None.
+   */
+  MaxPooling(const size_t kernelWidth,
+             const size_t kernelHeight,
+             const size_t strideWidth,
+             const size_t strideHeight,
+             const bool floor,
+             const size_t inputWidth,
+             const size_t inputHeight,
+             const std::tuple<size_t, size_t> padW,
+             const std::tuple<size_t, size_t> padH,
+             const std::string paddingType = "None");
```

Comment:
```suggestion
             const std::tuple<size_t, size_t> padW,
             const std::tuple<size_t, size_t> padH,
             const size_t inputWidth = 0,
             const size_t inputHeight = 0,
             const std::string paddingType = "None");
```
Input Width and Height should always come after padH and padW for the following reasons:
1. Consistency in code base.
2. User may want to pass padW / H but not input Width / Height as they might be set by visitors.
I'll try to make the suggestion everywhere.